openerp/addons/base/ir/ir_model.py

```diff
@@ -287,26 +290,68 @@ def _size_gt_zero_msg(self, cr, user, ids, context=None):
         ('size_gt_zero', 'CHECK (size>=0)',_size_gt_zero_msg ),
     ]
 
+    @api.onchange('ttype', 'model_id', 'relation')
+    def _onchange_ttype(self):
+        if self.ttype == 'many2many' and self.model_id and self.relation:
+            rel1 = self.relation.replace('.', '_')
+            rel2 = self.model_id.model.replace('.', '_')
+            self.relation_table = 'x_%s_%s_%s_rel' % (rel1, rel2, self.name)
```

Comment:
@KangOl should I generate a table name like `openerp.osv.fields.many2many._sql_names`?

    model1 = self.env[self.relation]
    model2 = self.env[self.model_id.model]
    self.relation_table = "x_%s_%s_rel" % tuple(sorted([model1._table, model2._table]))
    self.column1 = "%s_id" % model1._table
    self.column2 = "%s_id" % model2._table